kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiDatabaseMetaData.java

```diff
@@ -122,8 +126,50 @@ public ResultSet getCatalogs() throws SQLException {
         .build();
   }
 
+  private static final class ClientInfoPropertiesResultSet extends KyuubiMetaDataResultSet<Object> {
+    private static final String[] COLUMNS = {"NAME", "MAX_LEN", "DEFAULT_VALUE", "DESCRIPTION"};
+    private static final String[] COLUMN_TYPES = {"STRING", "INT", "STRING", "STRING"};
+
+    private static final Object[][] DATA = {
+      {"ApplicationName", 1000, null, null},
+      // Note: other standard ones include e.g. ClientUser and ClientHostname,
+      //       but we don't need them for now.
+    };
+    private int index = -1;
+
+    public ClientInfoPropertiesResultSet() throws SQLException {
+      super(Arrays.asList(COLUMNS), Arrays.asList(COLUMN_TYPES), null);
+      List<FieldSchema> fieldSchemas = new ArrayList<>(COLUMNS.length);
+      for (int i = 0; i < COLUMNS.length; ++i) {
+        fieldSchemas.add(new FieldSchema(COLUMNS[i], COLUMN_TYPES[i], null));
+      }
+      setSchema(new TableSchema(fieldSchemas));
+    }
+
+    @Override
+    public boolean next() throws SQLException {
+      if ((++index) >= DATA.length) return false;
+      row = Arrays.copyOf(DATA[index], DATA[index].length);
+      return true;
+    }
+
+    public <T> T getObject(String columnLabel, Class<T> type) throws SQLException {
+      for (int i = 0; i < COLUMNS.length; ++i) {
+        if (COLUMNS[i].equalsIgnoreCase(columnLabel)) return getObject(i, type);
+      }
+      throw new SQLException("No column " + columnLabel);
+    }
+
+    @SuppressWarnings("unchecked")
+    public <T> T getObject(int columnIndex, Class<T> type) throws SQLException {
+      // TODO: perhaps this could use a better implementation... for now even the Hive query result
```

Comment:
Is there a definite need here? IMO, you couldn't assume anything about the users here.

Reply:
I noticed [KyuubiQueryResultSet](https://github.com/apache/incubator-kyuubi/blob/93f13ef662f497a0e02281cf09cbf8cd0d8577fd/kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiQueryResultSet.java#L430) does not implement this method, shall I just add 
`throw new SQLFeatureNotSupportedException("Method not supported"); `instead ?